components/apimgt/org.wso2.carbon.apimgt.rest.api.util/src/main/java/org/wso2/carbon/apimgt/rest/api/util/interceptors/quotaLimit/ResourceQuotaLimitInterceptor.java

```diff
@@ -0,0 +1,116 @@
+    @Override
+    public void handleMessage(Message message) {
+        Message message1 = message;
+        try {
+            if (getQuotaLimitEnabled() && getToBeQuotaLimited(message) && getIsRegularAPICreation(message)) {
+                Map<String, String> queryParamsMap = UrlUtils.parseQueryString(
+                        (String) message.get(QuotaLimitInterceptorConstants.QUERY_PARAM_STRING));
+                if (queryParamsMap.containsKey(QuotaLimitInterceptorConstants.QUERY_PARAM_ORGANIZATION_ID)) {
+                    String organizationId = queryParamsMap.get(QuotaLimitInterceptorConstants.QUERY_PARAM_ORGANIZATION_ID);
+                    String userId = QuotaLimitInterceptorConstants.QUOTA_LIMIT_USERID;
+                    String resourceType = QuotaLimitInterceptorConstants.QUOTA_LIMIT_RESOURCE_TYPE;
+                    try {
+                        ResourceQuotaLimiter quotaLimiter = APIUtil.getResourceQuotaLimiter();
+                        if (quotaLimiter.getAPIQuotaLimitStatus(organizationId, userId, resourceType)) {
+                            Response response = Response.status(Response.Status.TOO_MANY_REQUESTS).build();
+                            message.getExchange().put(Response.class, response);
+                        }
+                    } catch (APIManagementException e) {
+                        e.printStackTrace();
+                    }
+                }
+                return;
+            }
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    private boolean getIsRegularAPICreation(Message message) {
+        boolean isRegularAPICreation = message.containsKey(QuotaLimitInterceptorConstants.APPLICATION_KEY);
+        return !isRegularAPICreation;
+    }
+
+    private boolean getQuotaLimitEnabled() {
+        APIManagerConfiguration configurations = ServiceReferenceHolder.getInstance().
+                getAPIManagerConfigurationService().getAPIManagerConfiguration();
+        boolean isQuotaLimitEnabled = Boolean.parseBoolean(configurations.getFirstProperty(APIConstants.API_QUOTA_LIMIT_ENABLE));
+        return isQuotaLimitEnabled;
+    }
+
+    private boolean getToBeQuotaLimited(Message message) throws IOException {
+        String httpMethod = (String) message.get(Message.HTTP_REQUEST_METHOD);
+        String matchingPath = (String) message.get(QuotaLimitInterceptorConstants.PATH_TO_MATCH_SLASH);
+
+        boolean isCreateAPI = QuotaLimitInterceptorConstants.HTTP_POST.equals(httpMethod) &&
+                QuotaLimitInterceptorConstants.API_FROM_SCRATCH_PATH.equals(matchingPath);
+        boolean isCreateVersion = QuotaLimitInterceptorConstants.HTTP_POST.equals(httpMethod) &&
+                QuotaLimitInterceptorConstants.NEW_API_VERSION_PATH.equals(matchingPath);
+        boolean isImportAPI = QuotaLimitInterceptorConstants.HTTP_POST.equals(httpMethod) &&
+                QuotaLimitInterceptorConstants.IMPORT_OPENAPI_PATH.equals(matchingPath);
```

Comment:
Can we move this assignments inside a if condition which checks `QuotaLimitInterceptorConstants.HTTP_POST.equals(httpMethod)` since this is duplicated. Or else if it reduces readability, we can introduce a new local var `isPostRequest` and use it here.